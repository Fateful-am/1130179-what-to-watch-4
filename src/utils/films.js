export const MovieInfoMode = {
  CARD: `card`,
  PROMO: `promo`,
  OVERVIEW: `overview`,
  DETAIL: `detail`,
  REVIEWS: `reviews`
};

export const getMovies = (sourceMovies, sliceFrom, count) => {
  return sourceMovies.slice(sliceFrom, sliceFrom + count).map((movie) => {
    return {
      id: movie.id,
      title: movie.title,
      posterImage: movie.posterImage,
    };
  });
};

export const getMoviePromoMode = (sourceMovies, index) => {
  const movie = sourceMovies[index];
  return {
    id: movie.id,
    title: movie.title,
    genre: movie.genre,
    posterImage: movie.posterImage,
    backgroundImage: movie.backgroundImage,
    released: movie.released
  };
};
