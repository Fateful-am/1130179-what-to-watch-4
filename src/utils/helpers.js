import moment from 'moment';

export const convertToImageName = (movieName) => {
  return movieName.toLowerCase().split(` `).join(`-`).split(`:`).join(``);
};

export const getMovieById = (movies, id) => {
  const movieIndex = movies.findIndex((movie) => movie.id === id);
  if (movieIndex > -1) {
    return movies[movieIndex];
  }
  return null;
};

export const formatDuration = (duration) => {
  if (duration < 60) {
    return moment.utc(moment.duration(duration, `minutes`).asMilliseconds()).format(`m[m]`);
  }
  return moment.utc(moment.duration(duration, `minutes`).asMilliseconds()).format(`H[h] m[m]`);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};


export const smallMovieCards = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`
];
