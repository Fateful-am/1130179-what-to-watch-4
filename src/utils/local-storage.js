import {extend} from './helpers';

const ItemsType = {
  AUTH_STATUS: `authStatus`,
  LAST_URL: `lastUrl`,
};

export default class LocalStorage {
  constructor(key, storage) {
    this._storage = storage;
    this._storeKey = key;
  }

  _getItems() {
    try {
      return JSON.parse(this._storage.getItem(this._storeKey)) || {};
    } catch (err) {
      return {};
    }
  }

  _getItem(item) {
    const store = this._getItems();
    return store[item];
  }

  _setItem(item, value) {
    const store = this._getItems();
    this._storage.setItem(
        this._storeKey,
        JSON.stringify(extend(store, {
          [item]: value,
        }))
    );
  }

  getAuthStatus() {
    return this._getItem(ItemsType.AUTH_STATUS);
  }

  setAuthStatus(authStatus) {
    this._setItem(ItemsType.AUTH_STATUS, authStatus);
  }

  getLastUrl() {
    return this._getItem(ItemsType.LAST_URL) || `/`;
  }

  setLastUrl(url) {
    this._setItem(ItemsType.LAST_URL, url);
  }
}
