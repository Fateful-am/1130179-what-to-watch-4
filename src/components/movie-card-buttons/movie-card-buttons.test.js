import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {TEST_DATA} from '../../utils/test-data';
import MovieCardButtons from './movie-card-buttons';
import NameSpace from '../../reducer/name-space';

const mockStore = configureStore([]);

it(`Render MovieCardButtons`, () => {
  const store = mockStore({
    [NameSpace.MOVIE]: TEST_DATA.initialStoreMovieState,
    [NameSpace.DATA]: TEST_DATA.initialStoreDataState,
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <MovieCardButtons />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render MovieCardButtons with AddReviewButton`, () => {
  const store = mockStore({
    [NameSpace.MOVIE]: TEST_DATA.initialStoreMovieState,
    [NameSpace.DATA]: TEST_DATA.initialStoreDataState,
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <MovieCardButtons>
            <a href="#" className="btn movie-card__button">Add review</a>
          </MovieCardButtons>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
