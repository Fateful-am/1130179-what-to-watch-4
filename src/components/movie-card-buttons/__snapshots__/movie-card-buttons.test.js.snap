// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCardButtons 1`] = `
<div
  className="movie-card__buttons"
>
  <button
    className="btn btn--play movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="none"
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <path
        clipRule="evenodd"
        d="M0 0L19 9.5L0 19V0Z"
        fill="#EEE5B5"
        fillRule="evenodd"
      />
    </svg>
    <span>
      Play
    </span>
  </button>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
</div>
`;

exports[`Render MovieCardButtons with AddReviewButton 1`] = `
<div
  className="movie-card__buttons"
>
  <button
    className="btn btn--play movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="none"
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <path
        clipRule="evenodd"
        d="M0 0L19 9.5L0 19V0Z"
        fill="#EEE5B5"
        fillRule="evenodd"
      />
    </svg>
    <span>
      Play
    </span>
  </button>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  <a
    className="btn movie-card__button"
    href="#"
  >
    Add review
  </a>
</div>
`;
